src/raftstore/store/snap.rs

```diff
@@ -807,6 +809,23 @@ mod v2 {
             .and_then(|_| check_file_checksum(path, expected_checksum))
     }
 
+    fn check_compression_available(compression: DBCompressionType) -> bool {
+        let compressions: Vec<DBCompressionType> = supported_compression();
+        compressions.contains(&compression)
+    }
+
+    fn get_fastest_compression() -> DBCompressionType {
+        // Zlib and bzip2 are too slow.
+        let compression_priority =
+            [DBCompressionType::DBLz4, DBCompressionType::DBZstd, DBCompressionType::DBSnappy];
+        for compression in &compression_priority {
+            if check_compression_available(*compression) {
+                return *compression;
+            }
+        }
+        DBCompressionType::DBNo
```

Comment:
for avoiding deref, maybe 

    for &val in &coll {}

is better.

For above logic, there's a brief one-liner:

    compressions.find(check_available).unwrap_or(::DBNo)